src/Build/BackEnd/Shared/ConfigurationMetadata.cs

```diff
@@ -58,6 +58,17 @@ public ConfigurationMetadata(string projectFullPath, PropertyDictionary<ProjectP
             _toolsVersion = MSBuildConstants.CurrentToolsVersion;
             _globalProperties = globalProperties;
         }
+        public ConfigurationMetadata(string projectFullPath, PropertyDictionary<ProjectPropertyInstance> globalProperties, string previousPlatform, string previousPlatformLookupTable)
+        {
+            ErrorUtilities.VerifyThrowArgumentLength(projectFullPath, nameof(projectFullPath));
+            ErrorUtilities.VerifyThrowArgumentNull(globalProperties, nameof(globalProperties));
+
+            _projectFullPath = projectFullPath;
```

Comment:
Instead of repeating, just call `this(projectFullPath, globalProperties)`?

Reply:
unfortunately that is not supported

Reply:
Sure it is?

```xml
        public ConfigurationMetadata(string projectFullPath, PropertyDictionary<ProjectPropertyInstance> globalProperties, string previousPlatform, string previousPlatformLookupTable)
            : this(projectFullPath, globalProperties)
        {
            PreviousPlatform = previousPlatform;
            PreviousPlatformLookupTable = previousPlatformLookupTable;
        }
```

Reply:
This looks good to me too; is there a reason this can't work?